packages/forms/src/core/utils/transform-utils.tsx

```diff
@@ -84,3 +84,22 @@ export function getTransformers(formDefinition: IFormDefinition): TransformItem[
 
   return ret
 }
+
+/** Remove values for inputs that are hidden  */
+export function unsetHiddenInputsValues(
+  formDefinition: IFormDefinition,
+  values: Record<string, any>,
+  metadata?: any
+): Record<string, any> {
+  const flattenInputs = flattenInputsRec(formDefinition.inputs)
+
+  const retValues = cloneDeep(values)
```

Comment:
this could get expensive for large forms. how frequently do we trigger this?
could we update fields in place?